tests/ast-alignment/fixtures-to-test.ts

```diff
@@ -508,6 +503,18 @@ let fixturePatternConfigsToTest = [
     ]
   }),
 
+  createFixturePatternConfigFor('typescript/declare', {
+    fileType: 'ts',
+    ignore: [
+      /**
+       * AST difference
+       */
+      'type-alias',
```

Comment:
Any reason why we did not align to the babel AST?

Reply:
https://github.com/JamesHenry/typescript-estree/pull/44

Reply:
i wanted to keep it cleaner, and easier to review

Reply:
I think it'd be better to be combined them into the same PR since this change does not seem to be good without #44 and this repo uses semantic-release (which means there will be an unstable version between this PR and #44). You could use 2 commits to keep it clean.

Reply:
hmm i have a lot more changes and i was trying to split them to smaller PR's but if you guys feel like i should merge this with #44 let be it, i will do merge now:

------

more changes will come letter: eg. https://github.com/armano2/typescript-estree/commit/77ef75a78dbf364a9897bebdb1e7e70769073262